src/VisionModel/vgg.jl

```diff
@@ -0,0 +1,92 @@
+include("convnb.jl")
+
+export vgg11, vgg11bn, vgg13, vgg13bn, vgg16, vgg16bn, vgg19, vgg19bn
+
+const configs = Dict(:A => [(64,1) (128,1) (256,2) (512,2) (512,2)],
+                     :B => [(64,2) (128,2) (256,2) (512,2) (512,2)],
+                     :D => [(64,2) (128,2) (256,3) (512,3) (512,3)],
+                     :E => [(64,2) (128,2) (256,4) (512,4) (512,4)])
+
+# Build a VGG block
+#  ifilters: number of input filters
+#  ofilters: number of output filters
+#  batchnorm: add batchnorm (see below for the problem of biases in Conv)
+function vgg_block(ifilters, ofilters, depth, batchnorm)
+  k = (3,3)
+  p = (1,1)
+  i = Flux.glorot_uniform
+  layers = []
+  for l in 1:depth
+    if batchnorm
+      # Conv with BatchNorm must have no biases, not possible with Flux v0.10.4, however it seems
+      # available in Flux#master
+      push!(layers, ConvNB(k, ifilters=>ofilters, pad=p, init=i))
+      push!(layers, BatchNorm(ofilters, relu))
+    else
+      push!(layers, Conv(k, ifilters=>ofilters, relu, pad=p, init=i))
+    end
+    ifilters = ofilters
+  end
+  return layers
+end
+
+# Build convolutionnal layers
+#  config: :A (vgg11) :B (vgg13) :D (vgg16) :E (vgg19)
+#  inchannels: number of channels in input image (3 for RGB)
+function convolutionnal_layers(config, batchnorm, inchannels)
+  layers = []
+  ifilters = inchannels
+  for c in configs[config]
+    push!(layers, vgg_block(ifilters, c..., batchnorm)...)
+    push!(layers, MaxPool((2,2)))
+    ifilters, _ = c
+  end
+  return Chain(layers...)
```

Comment:
I would prefer these to be a single `Chain` with both the conv and classifier layers. So can you return `layers` here?